src/hooks/useFhirClient.tsx

```diff
@@ -0,0 +1,130 @@
+import { useQuery, useMutation } from 'react-query';
+import { Bundle, Observation } from 'fhir/r3';
+import formatISO from 'date-fns/formatISO';
+import { useHttpClient } from './useHttpClient';
+import { useActiveAccount } from './useActiveAccount';
+import { useActiveProject } from './useActiveProject';
+import merge from 'deepmerge';
+
+type ResourceTypes = {
+  Observation: Observation;
+};
+type ResourceType = Observation;
+
+type QueryParams = {
+  resourceType: keyof ResourceTypes;
+  pageSize?: number;
+};
+
+type DeleteParams = {
+  id: string;
+  resourceType: keyof ResourceTypes;
+};
+
+export function useFhirClient() {
+  const { httpClient } = useHttpClient();
+  const { accountHeaders } = useActiveAccount();
+  const { activeProject, activeSubjectId } = useActiveProject();
+
+  const useSearchResourcesQuery = (queryParams: QueryParams) => {
```

Comment:
Should we use the lazy query [pattern](https://tanstack.com/query/v4/docs/react/guides/disabling-queries#lazy-queries) instead of returning a new function every time? Then we could wrap these in callbacks and not make them their own hooks.

Reply:
Yes, absolutely.  As mentioned [here](https://github.com/lifeomic/react-native-sdk/pull/113#discussion_r1158936177), I started with these as callbacks as-is.  Everything worked fine except lint, which pointed out that's a fundamental [violation](https://reactjs.org/link/invalid-hook-call) of rules of hooks.  But yea, let's definitely iterate on this over time to find the right balance of where the `useQuery` goes, how its consumed upstream (directly or indirectly), etc.